src/dtos/inputs/user-input.ts

```diff
@@ -0,0 +1,21 @@
+import { IsEmail, Matches, MaxDate, MinLength } from 'class-validator';
+import { Field, InputType } from 'type-graphql';
```

Comment:
quais motivos da decisão de utilizar `class-validator` e `type-graphql`?

Reply:
A decisão de utilizar o `type-graphql` foi pela sua facilidade e integração com TypeScript, ao escolher uma abordagem code-first o `type-graphql` me ajuda a não ter que criar um tipo do graphql e um do typescript, assim eu não preciso "duplicar" esse código em nenhum lugar, já que apenas utilizando os decorators o próprio type-grapql vai se encarregar de gerar o schema in-memory. Outro ponto pra minha decisão foi a facilidade de integração com outras ferramentas, como o typedi, permitindo eu passar o container direto na criação do schema e também o próprio class validator que para abilitar a verificação basta mudar a chave `validate` para true na definição do schema. Eu poderia utilizar uma ferramente de schema como o zod ou o yup mas optei pelo class validator por esses motivos: integração fácil com o type-grapql, recomendação da documentação do type-grapql e pelo padrão decorators que faz eu não precisar escrever um schema completo para a validação, como é no zod e no yup.